test/configCases/hash-length/output-filename/test.config.js

```diff
@@ -1,17 +1,37 @@
 var fs = require("fs");
 require("should");
 
+var verifyFilenameLength = function (files, regex, expectedNameLength) {
+	for(var j = 0, file; j < files.length; j++) {
+		file = files[j];
+		if (regex.test(file)) {
+			file.should.match(new RegExp("^.{" + expectedNameLength + "}$"));
+			return "./" + file;
+		}
+	}
+}
+
 module.exports = {
 	findBundle: function(i, options) {
 		var files = fs.readdirSync(options.output.path);
+
 		var expectedNameLength = options.amd.expectedFilenameLength;
-		var bundleDetect = new RegExp("^bundle" + i, "i");
-		for(var j = 0, file; j < files.length; j++) {
-			file = files[j];
-			if (bundleDetect.test(file)) {
-				file.should.match(new RegExp("^.{" + expectedNameLength + "}$"));
-				return "./" + file;
-			}
+		var bundleDetects = [{
+			regex: new RegExp("^bundle" + i, "i"),
+			expectedNameLength: options.amd.expectedFilenameLength
+		}, {
+			regex: new RegExp("^0.bundle" + i, "i"),
+			expectedNameLenght: options.amd.expectedChunkFilenameLenght
+		}
+		]
+
+		var bundleDetect;
+		for (bundleDetect of bundleDetects) {
+			return verifyFilenameLength(
```

Comment:
separate the validation loop from the searching loop. Elsewise you only validate file until one is found.
Don't return on undefined.